deployment/bases/solr/deployment.yaml

```diff
@@ -0,0 +1,145 @@
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: solr-data
+  labels:
+    app.kubernetes.io/name: solr
+    app.kubernetes.io/instance: s-enda
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: solr-config
+  labels:
+    app.kubernetes.io/name: solr
+    app.kubernetes.io/instance: s-enda
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: solr
+  annotations:
+    app.kubernetes.io/name: solr
+    app.kubernetes.io/instance: s-enda
+    app.kubernetes.io/version: v1.0
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: solr
+      app.kubernetes.io/instance: s-enda
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: solr
+        app.kubernetes.io/instance: s-enda
+    spec:
+      initContainers:
+        - name: init-db-directory
+          image: docker.io/alpine:latest
+          imagePullPolicy: Always
+          command:
+            - /bin/sh
+            - -c
+            - |
+              set -x
+              [ -s /mnt/schema.xml ] && echo "repo exists" && exit 0
+              apk update
+              apk add git
+              git clone https://github.com/magnarem/drupal8solr_configs.git /mnt
```

Comment:
Would have checked out a specific version, so you always start with what you know is a stable config. Maybe ensured that this always is defined in overrides to easily update the version in the future.